### ISSUE TITLE
Fabric ID is not updated during automated test runs ( TC-GRPKEY-2.1 , TC-SC-5.1, TC-ACL-2.7, TC-ACL-2.8)

### ISSUE THREAD

**Issue by raju-apple** [project-chip/certification-tool#50]
From chip-certification-tool created by [Shweta-UL](https://github.com/Shweta-UL): CHIP-Specifications/chip-certification-tool#972  

### Describe the bug

Fabric ID is not updated correctly for the listed test cases. However, during the manual test run we could notice the correct response from DUT.

### Log files

[TC-ACL-2.8_Manual_Last_Step_Pass.txt](https://github.com/CHIP-Specifications/chip-certification-tool/files/12662487/TC-ACL-2.8_Manual_Last_Step_Pass.txt)
[ACL_2_8_2023_09_15_15_37_29_Fail.log](https://github.com/CHIP-Specifications/chip-certification-tool/files/12662493/ACL_2_8_2023_09_15_15_37_29_Fail.log)
[ACL_2_7_2023_09_15_15_20_31_Fail.log](https://github.com/CHIP-Specifications/chip-certification-tool/files/12662502/
[TC_GRPKEY_2-1_2023_09_18_10_42_08.log](https://github.com/CHIP-Specifications/chip-certification-tool/files/12662544/TC_GRPKEY_2-1_2023_09_18_10_42_08.log)
ACL_2_7_2023_09_15_15_20_31_Fail.log)
[TC_SC_5-1_2023_09_18_12_56_06.log](https://github.com/CHIP-Specifications/chip-certification-tool/files/12662546/TC_SC_5-1_2023_09_18_12_56_06.log)
[TC-GRPKEY_2_1_Manual_Run_Log_Pass.txt](https://github.com/CHIP-Specifications/chip-certification-tool/files/12662550/TC-GRPKEY_2_1_Manual_Run_Log_Pass.txt)

**Comment by raju-apple** [comment]
@Shweta-UL  could you please confirm the prompt commands presented by TH UI were executed  after the following prefix:
`docker exec -it th-chip-tool `
**e.g.**  _docker exec -it th-chip-tool  ./chip-tool  pairing open-commissioning-window  1 1 400 2000 3841_

Please attach the chip-tool log as well(from the executions the got errors)

**Comment by raju-apple** [comment]
This issue has been reproduced also with TC_GRPKEY_2_1 which is an automated test case.
So we can eliminate the chip-tool log in this case.
With each re-run of the test the FabricIndex is increased by 1.

Please take a look at the attached log
[GRPKEY-2-1_2023_09_20_15_53_39.log](https://github.com/CHIP-Specifications/chip-certification-tool/files/12673903/GRPKEY-2-1_2023_09_20_15_53_39.log)

**Comment by raju-apple** [comment]
I ran some tests using `TC-GRPKEY-2.1` and the `chip-all-clusters-app` sample app. This were my findings:

- I was able to reproduce the problem when running the test in TH and also when running it manually, as long as I didn't reset the DUT after each test execution.
- If I reset the DUT by running `rm /tmp/chip_*` before starting the test I would not get this error, the FabricIndex would always be 1, as expected.
- So it seems like the DUT is the one keeping the FabricIndex information and incrementing it for each pairing (even if the last pairing has been removed).

I recorded the contents of the `/tmp/chip_vs` file form the DUT during the test execution and noticed that at the top, after commissioning the DUT, there are some lines that start with `f/1/` for the first execution and `f/2/` for the second execution. After reseting the DUT, these lines go back to `f/1/`. So I assume that this means FabricIndex.
[chip_kvs_evolution.txt](https://github.com/CHIP-Specifications/chip-certification-tool/files/12751093/chip_kvs_evolution.txt)

---

I'm not sure if this fabric increment is something defined by the spec or some implementation choice, I'd have to check.

Maybe the test case could have a pre-condition stating that the DUT must be factory reset before executing the test? Does that make sense?
CC @raju-apple
